src/server/test/hotkey_collector_test.cpp

```diff
@@ -0,0 +1,162 @@
+#include "server/hotkey_collector.h"
+
+#include <dsn/utility/rand.h>
+#include "pegasus_server_test_base.h"
+
+namespace pegasus {
+namespace server {
+
+static std::string generate_hash_key_by_random(bool is_hotkey, int probability = 100)
+{
+    if (is_hotkey && (dsn::rand::next_u32(100) < probability)) {
+        return "ThisisahotkeyThisisahotkey";
+    }
+    static const std::string chars("abcdefghijklmnopqrstuvwxyz"
+                                   "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                   "1234567890"
+                                   "!@#$%^&*()"
+                                   "`~-_=+[{]{\\|;:'\",<.>/? ");
+    std::string result;
+    for (int i = 0; i < 20; i++) {
+        result += chars[dsn::rand::next_u32(chars.size())];
+    }
+    return result;
+}
+
+TEST(hotkey_collector_test, get_bucket_id_test)
+{
+    int bucket_id = -1;
+    bucket_id = get_bucket_id(dsn::blob::create_from_bytes(generate_hash_key_by_random(false)));
```

Comment:
You can do a loop to generate a large number of 'keys' to check the functions's behavior, and you should check the bucket_id is hashed in range of [0, FLAGS_hotkey_buckets_num).